tests/gnmi/test_gnoi_system.py

```diff
@@ -35,6 +36,100 @@ def test_gnoi_system_time(duthosts, rand_one_dut_hostname, localhost):
     pytest_assert("time" in msg_json, "System.Time API did not return time")
 
 
+def test_gnoi_system_reboot(duthosts, rand_one_dut_hostname, localhost):
+    """
+    Verify the gNOI System Reboot API triggers a reboot and the device comes back online.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # Trigger reboot
+    ret, msg = gnoi_request(duthost, localhost, "Reboot", '{"method": 1}')
+    pytest_assert(ret == 0, "System.Reboot API reported failure (rc = {}) with message: {}".format(ret, msg))
+    logging.info("System.Reboot API returned msg: {}".format(msg))
+
+
+def test_gnoi_system_reboot_fail_invalid_method(duthosts, rand_one_dut_hostname, localhost):
+    """
+    Verify the gNOI System Reboot API fails with invalid method.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # Trigger reboot with invalid method
+    ret, msg = gnoi_request(duthost, localhost, "Reboot", '{"method": 99}')
+    pytest_assert(ret != 0, "System.Reboot API did not report failure with invalid method")
+
+
+def test_gnoi_system_reboot_when_reboot_active(duthosts, rand_one_dut_hostname, localhost):
+    """
+    Verify the gNOI System Reboot API fails if a reboot is already active.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # Trigger first reboot
+    ret, msg = gnoi_request(duthost, localhost, "Reboot", '{"method": 1}')
+    pytest_assert(ret == 0, "System.Reboot API reported failure (rc = {}) with message: {}".format(ret, msg))
+    logging.info("System.Reboot API returned msg: {}".format(msg))
+
+    # Trigger second reboot while the first one is still active
+    ret, msg = gnoi_request(duthost, localhost, "Reboot", '{"method": 1}')
+    pytest_assert(ret != 0, "System.Reboot API did not report failure when reboot is already active")
+
+
+def test_gnoi_system_reboot_status_immediately(duthosts, rand_one_dut_hostname, localhost):
+    """
+    Verify the gNOI System RebootStatus API returns the correct status immediately after reboot.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # Trigger reboot
+    ret, msg = gnoi_request(duthost, localhost, "Reboot", '{"method": 1, "message": "test"}')
+    pytest_assert(ret == 0, "System.Reboot API reported failure (rc = {}) with message: {}".format(ret, msg))
+    logging.info("System.Reboot API returned msg: {}".format(msg))
+
+    # Get reboot status
```

Comment:
>Get [](http://example.com/codeflow?start=6&length=3)

Could you query telemetry OTHERS/proc/uptime to ensure the device was really rebooted recently?